staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -204,6 +205,20 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 	c.HealthzChecks = append(c.HealthzChecks, healthz.NamedCheck("etcd", func(r *http.Request) error {
 		return healthCheck()
 	}))
+
+	if s.EncryptionProviderConfigFilepath != "" {
+		healthzConfigs, err := encryptionconfig.GetKMSPluginProbes(s.EncryptionProviderConfigFilepath)
+		if err != nil {
+			return err
+		}
+
+		for _, h := range healthzConfigs {
+			c.HealthzChecks = append(c.HealthzChecks, healthz.NamedCheck(fmt.Sprintf("kms-plugin-%s", h.Name), func(r *http.Request) error {
```

Comment:
this exposes the name of the configured kms in healthz output when unhealthy. are we comfortable with that, or should we just number them and include the name in the error message (which is not exposed via `/healthz`)?

Reply:
I originally implemented this as a number. @logicalhan suggested to use a more descriptive name to simplify troubleshooting which makes sense to me.

Reply:
including the name in the error message (which is logged and available via authenticated `/healthz/<check>` access) makes sense, but I would not expose that info via `/healthz`

Reply:
Switched to using a number to differentiate between providers and added a log message that includes provider's name.

Reply:
@liggitt, why is the name of configured kms too sensitive for healthz?

Reply:
The name of the KMS provider is something that is supplied by  users which, in some environments, may be considered to be PII.